packages/asset-uploader/src/targets/s3/handler.ts

```diff
@@ -0,0 +1,57 @@
+import { UploadHandler } from "@remix-run/node";
+import { PutObjectCommandInput, S3Client } from "@aws-sdk/client-s3";
+import { Upload } from "@aws-sdk/lib-storage";
+import ObjectID from "bson-objectid";
+
+import sharp from "sharp";
+import { ImagesUploadedSuccess } from "../../types";
+import {
+  getArrayBufferFromIterable,
+  getFilenameAndExtension,
+} from "../../helpers/array-buffer-helpers";
+
+export const s3UploadHandler: UploadHandler = async ({
+  data,
+  filename: baseFileName,
+  contentType,
+}) => {
+  if (!data) return;
+  const filename = baseFileName ?? ObjectID().toString();
+
+  const uint8Array = await getArrayBufferFromIterable(data);
+  const [name, extension] = getFilenameAndExtension(filename);
+  const key = `${name}_${Date.now()}.${extension}`;
+
+  const params: PutObjectCommandInput = {
+    Bucket: process.env.S3_BUCKET ?? "",
+    ACL: process.env.S3_ACL ?? "public-read",
+    Key: key,
+    Body: uint8Array,
+    ContentType: contentType,
+    Metadata: {
+      filename,
+    },
+  };
+
+  const client = new S3Client({
+    endpoint: process.env.S3_ENDPOINT ?? "",
```

Comment:
I think if any of those variables are missing, it should throw, fallback to empty strings will only obfuscate the error potentially, in all of those cases, potentially you could reuse a zod schema here and in env checks to make sure all env variables are present or its a hard failure

Reply:
It already is in the env check

Reply:
yes, but the logic suggests it can be empty when I read it I may not know it, also assume its a package with a public API the checks should happen here as well at the beginning and the logic should be clear that those are not optional

Reply:
I fixed it, let me know if its okay

Reply:
even after parsing, TS thinks that the env can be undefined I don't know why